assets/javascripts/discourse/initializers/qa-edits.js.es6

```diff
@@ -97,46 +76,24 @@ function initPlugin(api) {
       !post.reply_to_post_number &&
       !helper.widget.state.filteredRepliesShown
     ) {
-      const commentLinks = [];
-
-      if (helper.widget.attrs.canCreatePost) {
-        commentLinks.push(
-          helper.h("div.qa-comment-add", [
-            helper.attach("link", {
-              className: "qa-comment-add-link",
-              action: "openCommentComposer",
-              contents: () => I18n.t("qa.post.add_comment"),
-            }),
-          ])
-        );
-      }
-
       const postCommentsLength = post.comments?.length || 0;
 
       if (postCommentsLength > 0) {
         for (let i = 0; i < postCommentsLength; i++) {
           result.push(helper.attach("qa-comment", post.comments[i]));
         }
-
-        const mostPostCount = post.comments_count - postCommentsLength;
-
-        if (mostPostCount > 0) {
-          commentLinks.push(helper.h("span.qa-comment-seperator"));
-
-          commentLinks.push(
-            helper.h("div.qa-comment-show-more", [
-              helper.attach("link", {
-                className: "qa-comment-show-more-link",
-                action: "toggleFilteredRepliesView",
-                contents: () =>
-                  I18n.t("qa.post.show_comment", { count: mostPostCount }),
-              }),
-            ])
-          );
-        }
       }
 
-      result.push(helper.h("div.qa-comment-link", commentLinks));
+      if (helper.widget.attrs.canCreatePost) {
+        result.push(
+          helper.attach("qa-comment-menu", {
+            id: post.id,
+            moreCommentCount: post.comments_count - postCommentsLength,
+            lastPostNumber:
+              post.comments[post.comments.length - 1]?.post_number,
```

Comment:
When I post an answer locally, I get a JS error pointing to this line: 

`Uncaught TypeError: Cannot read properties of undefined (reading 'length')`

Reply:
Thank you for catching this, I've fixed it.